### PR TITLE
Change new([]*...) to slice zero value

### PR DESCRIPTION
**Pull request by gmlewis**
This change is a proposal and is open for discussion.

Change-Id: I62da2980c76d6220ce2ef12e6a1e64b60769ab01

### CODE REVIEW

File: github/repos_releases.go [hunk 1]
```diff
@@ -188,12 +188,12 @@ func (s *RepositoriesService) ListReleaseAssets(owner, repo string, id int, opt
 		return nil, nil, err
 	}
 
-	assets := new([]*ReleaseAsset)
-	resp, err := s.client.Do(req, assets)
+	var assets []*ReleaseAsset
+	resp, err := s.client.Do(req, &assets)
 	if err != nil {
 		return nil, resp, nil
```
Comment (shurcooL) [inline github/repos_releases.go]: I spotted this. It also happens in one more place in this file. See https://github.com/google/go-github/pull/63#pullrequestreview-19332985.

Do you want to include a fix for that in this PR @gmlewis?

File: github/integration.go [hunk 2]
```diff
@@ -28,11 +28,11 @@ func (s *IntegrationsService) ListInstallations(opt *ListOptions) ([]*Installati
 	// TODO: remove custom Accept header when this API fully launches.
 	req.Header.Set("Accept", mediaTypeIntegrationPreview)
 
-	i := new([]*Installation)
+	var i []*Installation
 	resp, err := s.client.Do(req, &i)
```
Comment (shurcooL) [inline github/integration.go]: I wonder if this even worked before, given the double pointer given to `client.Do`, i.e., `**[]*Installation`.

A quick test reveals that it should be fine, see https://play.golang.org/p/O5lq1RfseF. But it's still nice to fix to have simpler code.

File: github/repos_releases.go [hunk 3]
```diff
@@ -188,12 +188,12 @@ func (s *RepositoriesService) ListReleaseAssets(owner, repo string, id int, opt
 		return nil, nil, err
 	}
 
-	assets := new([]*ReleaseAsset)
-	resp, err := s.client.Do(req, assets)
+	var assets []*ReleaseAsset
+	resp, err := s.client.Do(req, &assets)
 	if err != nil {
-		return nil, resp, nil
+		return nil, resp, err
```
Comment (gmlewis) [inline github/repos_releases.go]: Fixed err return here. See https://github.com/google/go-github/pull/63#pullrequestreview-19332985

File: github/repos_releases.go [hunk 4]
```diff
@@ -210,7 +210,7 @@ func (s *RepositoriesService) GetReleaseAsset(owner, repo string, id int) (*Rele
 	asset := new(ReleaseAsset)
 	resp, err := s.client.Do(req, asset)
 	if err != nil {
-		return nil, resp, nil
+		return nil, resp, err
```
Comment (gmlewis) [inline github/repos_releases.go]: Fixed err return here. See https://github.com/google/go-github/pull/63#pullrequestreview-19332985

File: github/repos_releases.go [hunk 5]
```diff
@@ -248,7 +248,7 @@ func (s *RepositoriesService) DownloadReleaseAsset(owner, repo string, id int) (
 		if !strings.Contains(err.Error(), "disable redirect") {
 			return nil, "", err
 		}
-		return nil, loc, nil
+		return nil, loc, nil // Intentionally return no error with valid redirect URL.
```
Comment (gmlewis) [inline github/repos_releases.go]: Added comment here. See https://github.com/google/go-github/pull/312#commitcomment-20680837

**Comment by gmlewis** [comment]
Thank you, @shurcooL! I believe I addressed your comments.

**Comment by gmlewis** [comment]
Other readers - please note that the general handling of `return ..., nil` instead of `return ..., err` when `err == nil` will be handled in a separate PR.  I'll file an issue.

**Comment by gmlewis** [comment]
Yes, by "general handling", I mean all other instances.
Sorry for the auto-squash... the tools that I use perform this on my behalf. I'll try to remember to stop using them.  They were a requirement for working with the Go repos I worked on.

**Comment by gmlewis** [comment]
Merging.

**Comment by shurcooL** [comment]
> Yes, by "general handling", I mean all other instances.

Gotcha. 👍

> Sorry for the auto-squash... the tools that I use perform this on my behalf. I'll try to remember to stop using them. They were a requirement for working with the Go repos I worked on.

No problem, and thanks. I hope I'm not inconveniencing you too much. If this were Gerrit, it would keep track of your individual pushes so I could always see a diff between patch sets, but GitHub doesn't do that.

**Pull request merged**